src/main/java/frc/robot/states/AmpState.java

```diff
@@ -0,0 +1,56 @@
+package frc.robot.states;
+
+import com.pathplanner.lib.auto.AutoBuilder;
+import com.pathplanner.lib.util.GeometryUtil;
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.geometry.Transform2d;
+import edu.wpi.first.units.Units;
+import edu.wpi.first.wpilibj.DriverStation;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.Commands;
+import frc.robot.Constants;
+import frc.robot.swerve.SwerveDrive;
+import lib.Utils;
+
+public class AmpState implements ScoreState {
+
+    @Override
+    public Command driveToClosestOptimalPoint() {
+        return Commands.defer(
+                () -> {
+                    Pose2d ampPose = Constants.AMP_POSE;
+                    Pose2d botPose = SwerveDrive.getInstance().getBotPose();
+                    double robotRotation = botPose.getRotation().getRadians();
+                    boolean isGripperReversed =
+                            false; // TODO: replace with actual gripper.isForward
+                    var alliance = DriverStation.getAlliance();
+                    if (!alliance.isEmpty()) {
+                        if (alliance.get() == DriverStation.Alliance.Red) {
+                            ampPose = GeometryUtil.flipFieldPose(ampPose);
+                        }
+                    }
```

Comment:
You can use `isPresent()`. Also, thanks to [short-circuit evaluation](https://en.wikipedia.org/wiki/Short-circuit_evaluation), you can combine both conditions into one line (if the first condition is true, the second won't throw an exception, and if the first condition is false, the second one won't be checked):
```suggestion                    
                    if (alliance.isPresent() && alliance.get() == DriverStation.Alliance.Red) {
                        ampPose = GeometryUtil.flipFieldPose(ampPose);
                    }
```